### PR TITLE
enh: add logger level to server config

### PR DESCRIPTION
**Pull request by innerdvations**
### What does it do?

Loads log level from server.logger.config

### Why is it needed?

There was no documented way to change the log level, and loading it as a string from config/logger.js was not ideal

### How to test it?

Set server.logger.config or logger and it will use those values to pass to the logger setup for the log level

Default when nothing is set should be 'info'

### Related issue(s)/PR(s)

Let us know if this is related to any issue/pull request

### CODE REVIEW

**Review by Convly — COMMENTED** [review]
LGTM, but looks like the commitlint is broken (+ e2e tests, but I doubt you're guilty for those ones :sweat_smile: )

**Comment by innerdvations** [comment]
> LGTM, but looks like the commitlint is broken (+ e2e tests, but I doubt you're guilty for those ones 😅 )

TIL we have to use "enhancement" instead of "enh". I'll fix it in the squash, but we may want to update commitlint later too

File: packages/core/core/src/Strapi.ts [hunk 1]
```diff
@@ -215,7 +215,14 @@ class Strapi extends Container implements StrapiI {
     this.fs = createStrapiFs(this);
     this.eventHub = createEventHub();
     this.startupLogger = utils.createStartupLogger(this);
-    this.log = createLogger(this.config.get('logger', { level: 'info' }));
+
+    // We will continue to support 'logger' to prevent unnecessary deprecations but prioritize server.logger.level
+    // So we find: server.logger.level || logger || 'info'
+    const logLevel = this.config.get(
+      'server.logger.level',
```
Comment (innerdvations) [inline packages/core/core/src/Strapi.ts:222]:
```suggestion
      'server.logger',
```

**Comment by innerdvations** [comment]
@pwizla Very sorry for this! But I think you may have already written this documentation but the logger settings actually go in `logger: { config }` now instead of just `logger`

Additionally, I discovered we had documented in one place in v4 how to set [logger settings with config/logger.js](https://docs.strapi.io/dev-docs/configurations/middlewares#logger) which now needs to be marked as deprecated in favor of using server.js logger.config

**Comment by pwizla** [comment]
> @pwizla Very sorry for this! But I think you may have already written this documentation but the logger settings actually go in logger: { config } now instead of just logger

> Additionally, I discovered we had documented in one place in v4 how to set [logger settings with config/logger.js](https://docs.strapi.io/dev-docs/configurations/middlewares#logger) which now needs to be marked as deprecated in favor of using server.js logger.config

These are all updates for the v5 docs, @innerdvations, right?

**Comment by innerdvations** [comment]
> > @pwizla Very sorry for this! But I think you may have already written this documentation but the logger settings actually go in logger: { config } now instead of just logger
> 
> > Additionally, I discovered we had documented in one place in v4 how to set [logger settings with config/logger.js](https://docs.strapi.io/dev-docs/configurations/middlewares#logger) which now needs to be marked as deprecated in favor of using server.js logger.config
> 
> These are all updates for the v5 docs, @innerdvations, right?

Yep, all v5

**Pull request merged**